src/CVRF/cvrf_eval.c

```diff
@@ -0,0 +1,550 @@
+		}
+
+		while (oscap_string_iterator_has_more(product_ids)) {
+			product_id = oscap_string_iterator_next(product_ids);
+
+			xmlTextWriterStartElementNS(writer, NULL, BAD_CAST "Result", NULL);
+			xmlTextWriterWriteAttribute(writer, ATTR_PRODUCT_ID, BAD_CAST product_id);
+			if (cvrf_product_vulnerability_fixed(vuln, product_id)) {
+				xmlTextWriterWriteString(writer, BAD_CAST "FIXED");
+			}
+			else {
+				xmlTextWriterWriteString(writer, BAD_CAST "VULNERABLE");
+				vulnerable = true;
+			}
+			xmlTextWriterEndElement(writer);
+		}
+
+		if (vulnerable && cvrf_vulnerability_get_remediation_count(vuln) > 0) {
+			xmlTextWriterStartElementNS(writer, NULL, BAD_CAST "Remediations", NULL);
+			OSCAP_FOREACH(cvrf_remediation, e, cvrf_vulnerability_get_remediations(vuln), cvrf_remediation_export(e, writer);)
+			xmlTextWriterEndElement(writer);
+		}
+
+		oscap_string_iterator_free(product_ids);
+		xmlTextWriterEndElement(writer);
+		vulnerable = false;
+	}
+	cvrf_vulnerability_iterator_free(it);
+	xmlTextWriterEndElement(writer);
+
+	xmlTextWriterEndDocument(writer);
+	cvrf_model_eval_free(eval);
+	xmlFreeTextWriter(writer);
+	if (xmlGetLastError() != NULL)
+		oscap_setxmlerr(xmlGetLastError());
+}
+
+void get_cvrf_product_id_by_OS(struct cvrf_model_eval *eval, struct cvrf_model *model) {
+
+	const char *branch_product_id = NULL;
+	const char *relation_product_id = NULL;
+	struct cvrf_product_tree *tree = cvrf_model_get_product_tree(model);
+	struct cvrf_relationship *relation;
+	struct cvrf_product_name *full_name;
+	struct cvrf_branch *branch;
+
+	struct oscap_iterator *branches = cvrf_product_tree_get_branches(tree);
+	while (oscap_iterator_has_more(branches)) {
+		branch = oscap_iterator_next(branches);
+		branch_product_id = get_cvrf_product_id_from_branch(eval, branch);
+
+		if (branch_product_id != NULL)
+			break;
+	}
+	oscap_iterator_free(branches);
+	printf("Branch Product ID: %s\n", branch_product_id);
+	if (branch_product_id == NULL)
+		return;
+
+
+	struct cvrf_relationship_iterator *relationships = cvrf_product_tree_get_relationships(tree);
+	while (cvrf_relationship_iterator_has_more(relationships)) {
+		relation = cvrf_relationship_iterator_next(relationships);
+		if (!strcmp(branch_product_id, cvrf_relationship_get_relates_to_ref(relation))) {
+			full_name = cvrf_relationship_get_product_name(relation);
+			relation_product_id = cvrf_product_name_get_product_id(full_name);
+			printf("Relation Product ID: %s\n", relation_product_id);
+			oscap_stringlist_add_string(eval->product_ids, relation_product_id);
+		}
+	}
+	cvrf_relationship_iterator_free(relationships);
+}
+
+
+const char *get_cvrf_product_id_from_branch(struct cvrf_model_eval *eval, struct cvrf_branch *branch) {
+
+	const char *product_id = NULL;
+	const char *os_name = cvrf_model_eval_get_os_name(eval);
+	struct cvrf_branch *subbranch;
+	struct cvrf_product_name *full_name;
+
+	if (!strcmp(cvrf_branch_get_branch_type(branch), "Product Family")) {
```

Comment:
branch types can be arbitrary or is there a smaller set of types that are permitted? shouldn't this be an enum?